watertap/examples/flowsheets/oaro/oaro_multi.py

```diff
@@ -0,0 +1,797 @@
+        )
+        # add costing for ERD config
+        for erd in m.fs.EnergyRecoveryDevices.values():
+            erd.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    else:
+        erd_type_not_found(erd_type)
+
+    m.fs.recovered_pump_work = Expression(
+        expr=sum(
+            pyunits.convert(erd.work_mechanical[0], to_units=pyunits.kW)
+            for erd in m.fs.EnergyRecoveryDevices.values()
+        )
+    )
+    m.fs.net_pump_work = Expression(
+        expr=m.fs.total_pump_work + m.fs.recovered_pump_work
+    )
+
+    # additional parameters, variables or expressions ---------------------------------------------------------------------------
+    m.fs.oaro_min_pressure = Param(initialize=10e5, units=pyunits.Pa, mutable=True)
+    m.fs.oaro_max_pressure = Param(initialize=85e5, units=pyunits.Pa, mutable=True)
+    m.fs.ro_min_pressure = Param(initialize=10e5, units=pyunits.Pa, mutable=True)
+    m.fs.ro_max_pressure = Param(initialize=65e5, units=pyunits.Pa, mutable=True)
+    m.fs.recycle_pump_min_pressure = Param(
+        initialize=1e5, units=pyunits.Pa, mutable=True
+    )
+    m.fs.recycle_pump_max_pressure = Param(
+        initialize=10e5, units=pyunits.Pa, mutable=True
+    )
+
+    # process costing and add system level metrics
+    m.fs.costing.utilization_factor.fix(0.9)
+    m.fs.costing.factor_total_investment.fix(2)
+    m.fs.costing.factor_maintenance_labor_chemical.fix(0.03)
+    m.fs.costing.factor_capital_annualization.fix(0.1)
+    m.fs.costing.electricity_cost.set_value(0.07)
+    m.fs.costing.reverse_osmosis.factor_membrane_replacement.fix(0.15)
+    m.fs.costing.reverse_osmosis.membrane_cost.fix(30)
+    m.fs.costing.reverse_osmosis.high_pressure_membrane_cost.fix(50)
+    m.fs.costing.high_pressure_pump.cost.fix(53 / 1e5 * 3600)
+    m.fs.costing.energy_recovery_device.pressure_exchanger_cost.fix(535)
+
+    m.fs.costing.cost_process()
+    product_flow_vol_total = m.fs.product.properties[0].flow_vol
+    m.fs.costing.add_annual_water_production(m.fs.product.properties[0].flow_vol)
+    m.fs.costing.add_LCOW(m.fs.product.properties[0].flow_vol)
+    m.fs.costing.add_specific_energy_consumption(m.fs.product.properties[0].flow_vol)
+    m.fs.costing.add_specific_electrical_carbon_intensity(
+        m.fs.product.properties[0].flow_vol
+    )
+
+    # system water recovery
+    m.fs.volumetric_recovery = Var(
+        initialize=0.5,
+        bounds=(0, 1),
+        domain=NonNegativeReals,
+        units=pyunits.dimensionless,
+        doc="System Volumetric Recovery of Water",
+    )
+    m.fs.eq_volumetric_recovery = Constraint(
+        expr=m.fs.feed.properties[0].flow_vol_phase["Liq"] * m.fs.volumetric_recovery
+        == m.fs.product.properties[0].flow_vol_phase["Liq"]
+    )
+
+    m.fs.water_recovery = Var(
+        initialize=0.5,
+        bounds=(0, 1),
+        domain=NonNegativeReals,
+        units=pyunits.dimensionless,
+        doc="System Water Recovery",
```

Comment:
I'd prefer adding some notation relevant to "mass" recovery using whatever naming convention we used in LSRRO.